packages/extension-store-canner/src/lib/canner/persistenceStore.ts

```diff
@@ -0,0 +1,183 @@
+import {
+  APISchema,
+  ArtifactBuilderOptions,
+  PersistentStore,
+  TYPES,
+  VulcanExtensionId,
+  VulcanInternalExtension,
+} from '@vulcan-sql/core';
+import { inject } from 'inversify';
+import * as oas3 from 'openapi3-ts';
+import { createStorageService } from '../storageService';
+import { BaseStorageService } from '@canner/canner-storage';
+import { CannerStoreConfig, getEnvConfig } from '../config';
+import { geIndicatorFilesOfWorkspaces } from './utils';
+import { ArtifactIndicator } from './models';
+
+export interface RawBuiltInArtifact {
+  // key is source name, value is combined sql js executable code
+  templates: Record<string, string>;
+  schemas: APISchema[];
+  // specs content may be different according to the different spec generator, so we use the record type
+  specs: Record<string, any>;
+}
+
+export type BuiltInArtifact = RawBuiltInArtifact & {
+  specs: {
+    oas3: oas3.OpenAPIObject;
+  };
+};
+
+// key is workspaceSqlName, value is artifact buffer content
+interface WorkspaceArtifact {
+  workspaceSqlName: string;
+  artifact: RawBuiltInArtifact;
+}
+/**
+ * Used the string to identify the extension Id not by the enum "ArtifactBuilderProviderType" because the enum is define in the "core" package,
+ * and the canner extensions is only used for Canner integration, so we could add the "Canner" type in the "ArtifactBuilderProviderType",
+ * If we use the create an other enum and union "ArtifactBuilderProviderType", the enum is only has "Canner" type, so it seems define a other enum is unnecessary,
```

Comment:
Thanks for reviewing, it little different than your means after you simplify it.

We couldn't create another enum and union the original `ArtifactBuilderProviderType` to be a new enum type ( The formal term is enum inheritance )

The syntax does not allow if we need to make `Canner` an enum value, we create another enum, not it's unnecessary, because only has one enum value.